include/multipass/platform.h

```diff
@@ -91,7 +91,11 @@ std::unique_ptr<Process> make_sshfs_server_process(const SSHFSServerConfig& conf
 std::unique_ptr<Process> make_process(std::unique_ptr<ProcessSpec>&& process_spec);
 int symlink_attr_from(const char* path, sftp_attributes_struct* attr);
 
-std::function<int()> make_quit_watchdog(); // call while single-threaded; call result later, in dedicated thread
+// Creates a function that will wait for signals or until the passed function returns false.
+// The passed function is checked every timeout milliseconds.
+// If a signal is received the optional contains it, otherwise the optional is empty.
+std::function<std::optional<int>(const std::function<bool()>&)> make_quit_watchdog(
```

Comment:
Is there a reason to ever call this and then pass different functions to the result? If not, wouldn't it be clearer to tighten the closure as as much as possible early on and receive the periodic predicate as a parameter in this function? 

I mean, instead of a function that returns a function that receives a function, having a function that receives a function and returns a function. In essence, turn this into an adaptor, rather than something that returns an adaptor. WDYT?

Reply:
We don't ever pass different predicates and probably never will. But we can't pass the predicate at creation since this function has the precondition of being the only thread and the predicate we use only exists after we make another thread. Though it is possible, it'd be ugly

Reply:
Oh OK, got it.